base/applications/network/ftp/main.c

```diff
@@ -227,7 +227,10 @@ int main(int argc, const char *argv[])
 		(void) strcpy(home, pw->pw_dir);
 	}
 #endif
-        strcpy(home, "C:/");
+	if (GetEnvironmentVariableA("SystemDrive", home, ARRAYSIZE(home)))
+		strcat(home, "/");
+	else
+		strcpy(home, "C:/");
```

Comment:
Since this whole application is coded in POSIX way, please use instead `getenv()`:
```suggestion
        cp = getenv("SystemDrive");
        if ((cp != NULL) && *cp) {
                strcpy(home, cp);
                strcat(home, "/");
        } else {
                strcpy(home, "C:/");
        }
```
(I also fixed the indentation and kept the ambient C code style so as to match with the rest of the code.)

Reply:
The file uses tabs and you used spaces ;)

Reply:
oh yes indeed ^^ (actually a mixture)